process/economics/economicsData.go

```diff
@@ -431,6 +446,17 @@ func (ed *economicsData) EpochConfirmed(epoch uint32) {
 
 	ed.flagGasPriceModifier.Toggle(epoch >= ed.gasPriceModifierEnableEpoch)
 	log.Debug("economics: gas price modifier", "enabled", ed.flagGasPriceModifier.IsSet())
+	ed.setGasPriceModifierMetric()
+}
+
+func (ed *economicsData) setGasPriceModifierMetric() {
+	if check.IfNil(ed.statusHandler) {
+		log.Debug("cannot set GasPriceModifier metric - nil status handler")
+	}
+
+	if ed.flagGasPriceModifier.IsSet() {
+		ed.statusHandler.SetStringValue(core.MetricGasPriceModifier, fmt.Sprintf("%g", ed.gasPriceModifier))
```

Comment:
This will not work as expected in case of rollbacks. Suggestion to get rid of the if branch

Reply:
No, I can't remove the `if` because then it will be executed at every epoch, not just when the new gas price modifier comes in.